prohibit FIFOs for wordlist+fork and as pot files

AlekseyCherepanov opened the pull request:
- The first commit addresses #5000 as @solardiz suggested. Opening of file is done before `fstat`, so john waits other end of fifo to be opened and only then exits with error. I guess it is ok because user would be going to provide input, if we protect from intentional use.
- The second commit prohibits use of fifos as .pot files. Fifo as argument to `--loopback=` would be opened twice. So it would hang even with intentional use.
- Also I noticed that several warnings print 'name' without `path_expand()`. But 2 cases had `path_expand()` called earlier. The 2 fixed cases are not such.

I cannot say that I tested it well. I think there might be some consequences I could not foresee.

magnumripper commented on src/loader.c:
```diff
@@ -209,13 +209,20 @@ static void read_file(struct db_main *db, char *name, int flags,
-			if (S_ISDIR(file_stat.st_mode)) return;
+	if (stat(name, &file_stat)) {
+		if (flags & RF_ALLOW_MISSING)
+			if (errno == ENOENT) return;
```
Our current code style says we should put `return;` on next line. Same below.

magnumripper commented on src/loader.c:
```diff
@@ -209,13 +209,20 @@ static void read_file(struct db_main *db, char *name, int flags,
+	if (ldr_in_pot && S_ISFIFO(file_stat.st_mode)) {
+		if (john_main_process)
+			fprintf(stderr, "Error, cannot use FIFO as pot"
+				" file (%s)\n", path_expand(name));
```
Our current code style recommends to avoid breaking output strings even if it exceeds normal column width. Same for several other changes in this ~commit~ PR.

magnumripper commented:
> I cannot say that I tested it well. I think there might be some consequences I could not foresee.

The logic looks good to me

AlekseyCherepanov commented:
I fixed the style problems near my changes.

I used parentheses in `((flags & RF_ALLOW_DIR) && S_ISDIR(file_stat.st_mode))` because there was code with parentheses below already. But they are not necessary because `&&` has lower precedence than `&`.

Both styles are present in john:
```
$ grep '([a-zA-Z_0-9]\+ & [a-zA-Z_0-9]\+) &&' src/*.c
src/bench.c:		if ((flags & 0x200) && pass >= 2)
src/external.c:	if ((ext_flags & EXT_REQ_GENERATE) && !f_generate) {
[...]

$ grep '([a-zA-Z_0-9]\+ & [a-zA-Z_0-9]\+ &&' src/*.c
src/simd-intrinsics.c:	if (SSEi_flags & SSEi_REVERSE_STEPS && !(SSEi_flags & SSEi_CRYPT_SHA224))
src/simd-intrinsics.c:	if (SSEi_flags & SSEi_REVERSE_STEPS && !(SSEi_flags & SSEi_CRYPT_SHA384))
src/wpapcap2john.c:			if (l & 1 && l < 10)
```

solardiz commented on src/wordlist.c:
```diff
@@ -574,6 +575,15 @@ void do_wordlist_crack(struct db_main *db, const char *name, int rules)
+		if (options.fork && file_is_fifo) {
+			if (john_main_process)
+				fprintf(stderr, "Error, cannot use --fork "
+					"with FIFO as dictionary file.\n");
```
We normally say "wordlist", not "dictionary", and no need to wrap source code lines shorter than 132.

AlekseyCherepanov replied:
Oops, I forgot that I wrapped lines in `wordlist.c`.

I replicated 'dictionary' from other warning. Now both places are fixed.

solardiz reviewed (approved):
I've just made a nitpick comment.

Also, I wouldn't introduce the coding style changes as a separate commit - when those changes are limited to places touched by the previous commits, they can be right in those commits. However, this is also a nitpick - not strictly required to be "fixed", we can also merge as-is.

AlekseyCherepanov commented:
I rewrote commits to include changes of style for my changes and to near places unrelated to the changes.

Also I did not split lines shorter than 132. Example, new vs previous variants:
```c
				fprintf(stderr, "Warning: UTF-8 seen reading %s\n", path_expand(name));
// older
				fprintf(stderr, "Warning: UTF-8 seen reading %s\n",
					path_expand(name));
// oldest
				fprintf(stderr, "Warning: UTF-8 seen reading "
				        "%s\n", name);
```
(I noticed in "older" that I put tab to align arguments. It was not intentional.)

AlekseyCherepanov commented:
> AlekseyCherepanov force-pushed the fix26 branch from e6661da to f988634 2 hours ago
> Compare

Pretty convenient: after `git push -f` github shows a line with event. `force-pushed` and `Compare` button are links to comparisons between states, so it is possible to check cumulative change in addition to or instead of reviewing separate patches.

AlekseyCherepanov committed:
wordlist mode: prohibit --fork together with FIFO as wordlist

AlekseyCherepanov committed:
loader: prohibit FIFO as pot file

AlekseyCherepanov committed:
loader: add path_expand() for name in some warnings

AlekseyCherepanov committed:
wordlist, loader: fix code style and wording

AlekseyCherepanov commented:
I changed ` (%s)` to `: %s` in error message. I rewrote commits to include changes to code style only in my changes. Change for code style and `dictionary`->`wordlist` substitution in other places were extracted into a separate commit.
